### PR TITLE
Dashboard Only Shows Workers in its Cluster

### PR DESCRIPTION
**Pull request by mfitton**
## Why are these changes needed?
In the scenario that we have multiple Ray clusters with processes running on the same node, we want to make sure that each cluster's dashboard only contains metrics from the processes in its cluster.
The ray `reporter` process collects node-level metrics about ray workers, but some of the ray worker processes running on a given node could belong to other ray clusters (the way workers are currently identified is through a process name-prefix, and we will eventually want to make more structured). We could have, in theory, gotten a list of relevant worker process IDs via an RPC call; however this would have have added strain to the core system.
For the time being at least, I have placed the relevant changes on the front end. There is a concept now of `clusterWorkers`, which is a subset of the `workers` in a given node's stats. 
This `clusterWorkers` subset is calculated by looking at the process IDs of the workers in the Raylet stats that the reporter fetches for the dashboard.
The data is then used to filter the worker values that are being passed into various parts of the UI, such as node details table data and data aggregation rows.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/dashboard/client/src/pages/dashboard/node-info/NodeInfo.tsx [hunk 1]
```diff
@@ -12,12 +12,31 @@ import {
...
 import { StoreState } from "../../../store";
 import Errors from "./dialogs/errors/Errors";
 import Logs from "./dialogs/logs/Logs";
 import NodeRowGroup from "./NodeRowGroup";
 import TotalRow from "./TotalRow";
 
+const clusterWorkerPids = (
+  rayletInfo: RayletInfoResponse,
+): Map<string, Set<string>> => {
+  // Given a Raylet response, this extracts, per node, all the worker pids registered
```
Comment (edoakes) [inline python/ray/dashboard/client/src/pages/dashboard/node-info/NodeInfo.tsx:25]: I found this sort of hard to understand - maybe "Groups PIDs registered with each raylet by node IP address." ?

File: python/ray/dashboard/client/src/pages/dashboard/node-info/NodeInfo.tsx [hunk 2]
```diff
@@ -92,31 +111,39 @@ class NodeInfo extends React.Component<
       };
     } = {};
 
+    const clusterWorkerPidsByIp = clusterWorkerPids(rayletInfo);
```
Comment (edoakes) [inline python/ray/dashboard/client/src/pages/dashboard/node-info/NodeInfo.tsx:114]: High level comment describing what the following block is doing would be very helpful here

**Review by edoakes — APPROVED** [review]
LGTM, just some small suggestions

**Comment by mfitton** [comment]
@mehrdadn it looks like there is a new issue with the linting, at least on this branch:
```
fatal: Invalid symmetric difference expression 122353c392b1a829ece179f7518ad3322255fe3e...c5145d4e928727c73f06828c659f3c512d18045b
50
Traceback (most recent call last):
51
  File "/Users/runner/runners/2.169.1/work/ray/ray/ci/travis/determine_tests_to_run.py", line 60, in <module>
52
    files = list_changed_files(commit_range)
53
  File "/Users/runner/runners/2.169.1/work/ray/ray/ci/travis/determine_tests_to_run.py", line 29, in list_changed_files
54
    out = subprocess.check_output(command)
55
  File "/usr/local/Cellar/python@2/2.7.17_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 223, in check_output
56
    raise CalledProcessError(retcode, cmd, output=output)
57
subprocess.CalledProcessError: Command '['git', 'diff', '--name-only', '122353c392b1a829ece179f7518ad3322255fe3e...c5145d4e928727c73f06828c659f3c512d18045b', '--']' returned non-zero exit status 128
```

**Comment by mehrdadn** [comment]
@mfitton Try merging in master?

**Pull request merged**